ext/sqlite3/sqlite3.c

```diff
@@ -117,7 +117,10 @@ PHP_METHOD(sqlite3, open)
 	if (strlen(filename) != filename_len) {
 		return;
 	}
-	if (filename_len != sizeof(":memory:")-1 ||
+	if (filename_len == 0) {
+		fullpath = emalloc(1);
```

Comment:
I agree that the emalloc(1) is silly, but I'm not sure that `fullpath = NULL` would work (it does now, but that might be an implementation detail). The SQLite3 docs say "empty filename".

What about using a stack allocated empty string instead?